fastf1/tests/test_events.py

```diff
@@ -132,6 +132,46 @@ def test_event_schedule_get_by_name():
     assert schedule.get_event_by_name('test-test').EventName == 'test_test'
 
 
+def test_event_fuzzy_search():
+    # highest overlap case
+    schedule = fastf1.get_event_schedule(1979)
+    assert schedule.get_event_by_name(
+        "United States").EventName == "United States Grand Prix"
+    assert schedule.get_event_by_name(
+        "United States Grand Prix").EventName == "United States Grand Prix"
+    assert schedule.get_event_by_name(
+        "United States West").EventName == "United States Grand Prix West"
+    assert schedule.get_event_by_name(
+        "United States West Grand Prix").EventName == "United States Grand Prix West"
+    assert schedule.get_event_by_name(
+        "US West").EventName == "United States Grand Prix West"
+
+    # tests for common inputs
+    schedule = fastf1.get_event_schedule(2024)
+    # assert schedule.get_event_by_name(
+    #     "Saudi").EventName == "Saudi Arabian Grand Prix"
+    # assert schedule.get_event_by_name(
+    #     "Saudi GP").EventName == "Saudi Arabian Grand Prix"
```

Comment:
I would love to be able to handle these cases correctly since they are reasonable inputs. This does not seem possible due to interference from `Abu Dhabi Grand Prix`. At least we are emitting a warning to users now.

Reply:
What if you try to handle this in a way similar to the current ``driver_color`` and ``team_color`` functions? Before doing fuzzy matching, you add a check for partial exact matches. That way, these might work as well?

Reply:
I tried that at first but doing a partial ratio only would cause failures in the `United States Grand Prix` vs `United States Grand Prix West` edge case.

Solution now is to check for partial matches first and fall back to using the full ratio if:

- there is zero 100% match
- there are multiple 100% matches

Reply:
That seems like a good solution 👍